src/cartesi-machine.lua

```diff
@@ -257,8 +257,80 @@ where options are:
   --auto-reset-uarch-state
     automatically reset state after miroarchitecture halt
 
-  -i or --htif-console-getchar
-    run in interactive mode.
+  --htif-console-getchar
+    run in interactive mode
+
+  --virtio-console
+    add a VirtIO console device,
+    this console is more responsive than HTIF console and has support for terminal size
+
+  -i
+    run in an interactive mode using a VirtIO console device
+    same as --htif-console-getchar --virtio-console
+
+  --share-dir=<directory>
+    Add a VirtIO Plan 9 filesystem device sharing a host directory with the guest.
+
+    This argument can be specified multiple times, each filesystem device will have
+    a tag (e.g. vfs0, vfs1, ...) that keeps incrementing to be used with the mount command.
+
+    The filesystem can be mounted in the guest with the following command:
+      mount -t 9p vfs0 /mnt
+
+  --net-dev=user
+    Add a VirtIO network device using host user-space networking,
+    this effectively allows to use host network from inside the machine with ease.
+
+    This is a user-mode network,
+    meaning you don't need root privilege or any configuration in the host to use it,
+    in most case it should work out of the box.
+
+    While being of use, this mode has the following limitations:
+      - There is an additional an emulation layer of the TCP/IP stack, so there is an overhead.
+      - Not all IP protocols are emulated, but TCP and UDP should work.
+      - Host cannot connect to guest TCP ports.
+
+    The implementation uses libslirp TCP/IP emulator library.
+
+    To enable networking, execute the following commands inside the guest with root privilege:
+      ip link set dev eth0 up
+      ip addr add 10.0.2.1/24 dev eth0
+      ip route add default via 10.0.2.2 dev eth0
+      echo 'nameserver 10.0.2.3' > /etc/resolv.conf
+
+    The network settings configuration is fixed to the following:
+      Network:      10.0.2.0
+      Netmask:      255.255.255.0
+      Host/Gateway: 10.0.2.2
+      DHCP Start:   10.0.2.15
+      Nameserver:   10.0.2.3
+
+  --net-dev=<iface>
+    Add a VirtIO network device using host TUN/TAP interface,
+    this effectively allows to use host network from inside the machine.
+
+    This is more efficient and has less limitations than the user-space networking.
+
+    The following commands in the host with root privilege before starting the machine:
+      modprobe tun
+      ip link add br0 type bridge
+      ip tuntap add dev tap0 mode tap user $USER
+      ip link set dev tap0 master br0
+      ip link set dev br0 up
+      ip link set dev tap0 up
+      ip addr add 192.168.3.1/24 dev br0
+      sysctl -w net.ipv4.ip_forward=1
+      iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+
+    In the example above the host public internet interface is eth0,
+    but this depends in the host.
+
+    Finally start the machine with using --net-dev=tap0 and
```

Comment:
I would add these commands to the initialization too (and perhaps remove from the documentation here)